### DISCUSSION TITLE
How to make a cppflow::tensor from vector or array?

### DISCUSSION THREAD

**Discussion by DemO-O-On** [serizba/cppflow#248] [Q&A]
Hello, I'm a newbuy in C++ and I have a question: can I create cppflow::tensor from 2-dimensional vector or array? I've tried

```
std::vector<std::vector<int64_t>> data {{1,2,3}, {4,5,6}};

std::vector<int64_t> shape{2,3};

auto b = cppflow::tensor(data, shape);
```
but it dropped me
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Could not deduce type! type_name: St6vectorIxSaIxEE
```
What am I doing wrong?

**Comment by AntoneRoundy** [comment]
Here's the relevant code that I used to do something similar -- what worked for me was to first create a tensor using the TensorFlow C API, and then use that to create a CppFlow tensor object:

#include "cppflow/cppflow.h"
#include "tensorflow/c/tf_tensor.h"

int64_t dims[3];
TF_Tensor *pTensor;
cppflow::tensor *pCppFlowTensor;

dims[0] = 640; // image width
dims[1] = 480; // image height
dims[2] = 3; // color planes
pTensor = TF_AllocateTensor(TF_UINT8, dims, 3, dims[0] * dims[1] * dims[2] * sizeof(byte));
pCppFlowTensor = new cppflow::tensor(pTensor);
